packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx

```diff
@@ -0,0 +1,164 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { gql, useMutation, useQuery } from '@apollo/client';
+import TextField from '@material-ui/core/TextField';
+import { useEffect, useState } from 'react';
+import {
+  GQLInitialDirectEditElementLabelData,
+  GQLInitialDirectEditElementLabelInput,
+  TreeItemDirectEditInputProps,
+} from './TreeItemDirectEditInput.types';
+
+const renameTreeItemMutation = gql`
+  mutation renameTreeItem($input: RenameTreeItemInput!) {
+    renameTreeItem(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+const initialDirectEditElementLabeQuery = gql`
+  query initialDirectEditElementLabel($editingContextId: ID!, $representationId: ID!, $treeItemId: ID!) {
+    viewer {
+      editingContext(editingContextId: $editingContextId) {
+        representation(representationId: $representationId) {
+          description {
+            ... on TreeDescription {
+              initialDirectEditTreeItemLabel(treeItemId: $treeItemId)
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export const TreeItemDirectEditInput = ({
+  editingContextId,
+  treeId,
+  treeItemId,
+  refParent,
+  editingkey,
+  onClose,
+}: TreeItemDirectEditInputProps) => {
+  const [state, setState] = useState({
+    newLabel: '',
+    editingkey: editingkey,
+  });
+
+  const {
+    loading: initialLabelTreeItemLoading,
+    data: initialLabelTreeItemItemData,
+    error: initialLabelTreeItemItemError,
+  } = useQuery<GQLInitialDirectEditElementLabelData, GQLInitialDirectEditElementLabelInput>(
+    initialDirectEditElementLabeQuery,
+    {
+      variables: {
+        editingContextId: editingContextId,
+        representationId: treeId,
+        treeItemId: treeItemId,
+      },
+    }
+  );
+
+  useEffect(() => {
+    if (
+      !initialLabelTreeItemLoading &&
+      !initialLabelTreeItemItemError &&
+      initialLabelTreeItemItemData?.viewer.editingContext.representation.description.initialDirectEditTreeItemLabel
+    ) {
+      const initialLabel =
+        initialLabelTreeItemItemData?.viewer.editingContext.representation.description.initialDirectEditTreeItemLabel;
+      setState((prevState) => {
+        return { ...prevState, newLabel: initialLabel + state.editingkey };
+      });
+    }
+  }, [initialLabelTreeItemLoading, initialLabelTreeItemItemError, initialLabelTreeItemItemData]);
+
+  const [renameTreeItem, { loading: renameTreeItemLoading, data: renameTreeItemData, error: renameTreeItemError }] =
+    useMutation(renameTreeItemMutation);
+
+  useEffect(() => {
+    if (!renameTreeItemLoading && !renameTreeItemError && renameTreeItemData?.renameTreeItem) {
+      const { renameTreeItem } = renameTreeItemData;
+      if (renameTreeItem.__typename === 'SuccessPayload') {
+        onClose();
+      }
```

Comment:
- The loading is not useful, you may not be loaded and may not have errors and still not have any data so test using data only
- Consider errors properly and display them like in any other React components